wormhole-connect/src/hooks/useAmountValidation.ts

```diff
@@ -0,0 +1,109 @@
+import { amount as sdkAmount, routes } from '@wormhole-foundation/sdk';
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { QuoteResult } from 'routes/operator';
+import { RootState } from 'store';
+import { RouteState } from 'store/transferInput';
+
+type HookReturn = {
+  error?: string;
+  warning?: string;
+};
+
+type Props = {
+  balance?: string | null;
+  routes: RouteState[];
+  quotesMap: Record<string, QuoteResult | undefined>;
+  tokenSymbol: string;
+  isLoading: boolean;
+};
+
+export const useAmountValidation = (props: Props): HookReturn => {
+  const { amount } = useSelector((state: RootState) => state.transferInput);
+
+  // Min amount available
+  const minAmount = useMemo(
+    () =>
+      Object.values(props.quotesMap).reduce((minAmount, quoteResult) => {
+        if (quoteResult?.success) {
+          return minAmount;
+        }
+
+        const minAmountError = quoteResult?.error as routes.MinAmountError;
+
+        if (!minAmountError?.min) {
```

Comment:
this could be more concise and easier to follow. I'd first establish that `quoteResult` is defined, then use `instanceof`:

```
if (!quoteResult || quoteResult.success) {
  return minAmount;
}

if (quoteResult.error instanceof routes.MinAmountError) {
  // ... do the thing
} else {
  return minAmount;
}
```

`instanceof` handles undefined properties just fine:

<img width="212" alt="image" src="https://github.com/user-attachments/assets/cbc4ed86-4bce-4392-8486-9d9e8d1b3b4f">

Reply:
Indeed, that was my first implementation, but for some reason I consistently getting false negative results.
Here is the output of the error, MinQuoteError and instance of:
![Screenshot from 2024-09-17 01-01-25](https://github.com/user-attachments/assets/50304186-87bd-4440-b653-4b7205469aa0)

I suspect MinQuoteError is being transpiled/bundled to 2 different JS chunks, and since inctanceof does the check by reference, if fails. Tho, I'm not sure about it, didn't dug into it. I'll add a 'workaround' comment in code.

Reply:
oh got it. interesting.

one other thing we could do is add a little utility function like `isMinAmountError` which type-guards it. you could put it in `utils/sdkv2.ts`. could be similar to this: https://github.com/wormhole-foundation/wormhole-sdk-ts/blob/main/connect/src/types.ts#L137-L141

you would do that instead of casting using `as`. wdyt?